### ISSUE TITLE
Type mismatch error when passing a callback as an argument

### ISSUE THREAD

**Issue by Deirel** [HaxeFoundation/intellij-haxe#1105]
I encountered a "Type mismatch" error when passing a callback as an argument to the function. The error message states that the expected type is 'Void -> Bool', but the actual type is 'Bool'. 

I use the dev version of the plugin for IDEA 2022.3.3 built from the commit `5ea175f71663e6d457a4236f87e0fd8f735c5186`.

Here is the code to reproduce the issue:
```
class Test {
    public function new() {
        foo(callback);  // Type mismatch (Expected: 'Void->Bool' got: 'Bool')
    }

    function foo(callback:Void -> Bool):Void {
        trace(callback());
    }

    function callback():Bool {
        return true;
    }
}
```
It is interesting that if the function takes an argument of type `Array<Void -> Bool>`, there is no error:
```
foo2([callback]);  // It's ok

function foo2(callbacks:Array<Void -> Bool>):Void {...}
```
The same is if the argument type is defined as typedef:
```
foo3(callback);  // it's also ok

function foo3(callback:VoidToBool):Void {...}

typedef VoidToBool = Void -> Bool
```

**Comment by m0rkeulv** [comment]
Should be mostly working in version 1.4.1
There's still some work to be done with generics and function types, so it currently wont detect type mismatch when the type is a functionType and is used as a generic type

![image](https://github.com/HaxeFoundation/intellij-haxe/assets/3193925/8cb07f3d-666e-46c0-bf47-e5b50d1a1118)

**Comment by m0rkeulv** [comment]
should be  fully fixed in next release (1.4.2)
![image](https://github.com/HaxeFoundation/intellij-haxe/assets/3193925/d099ccba-2416-49c7-b7e1-d4cde1ac3efb)
